### ISSUE TITLE
Crossword hint on The Guardian doesn't work

### ISSUE THREAD

**Issue by The-Compiler** [The-Compiler/qutebrowser#1445]
When pressing `aa` on http://www.theguardian.com/ here:

![guardian](https://cloud.githubusercontent.com/assets/625793/14748945/f163cd94-08bc-11e6-9c22-b305a7302297.png)

We get to the "[This column will change your life](http://www.theguardian.com/lifeandstyle/2016/apr/22/oliver-burkeman-change-life-ethical-consumers)" article instead.

This might be fixed by @lahwaacz's changes, but it'd be nice to have a minimal testcase for it.

It seems like the link element is 0x0px?

![guardian2](https://cloud.githubusercontent.com/assets/625793/14749076/ade1b6c0-08bd-11e6-89af-725d96c99c7f.png)

cc @error800

**Comment by lahwaacz** [comment]
Yeah, this one works fine for me, but it's just a happy coincidence. There already are (will be) tests for `0x0` links (e.g. [this one](https://github.com/lahwaacz/qutebrowser/blob/hints_clicking/tests/integration/data/hints/html/nested_block_style.html)), but now I see that the behaviour implemented in #1311 does not work as expected when the nested element has non-zero `margin-top` attribute. For example a test with the following link does not work:

```html
<a href="/data/hello.txt">
    <span style="display: block; margin-top: 3rem;">
        link
    </span>
</a>
```

When the top margin is higher than the nested element itself, the click goes to the margin. On the Guardian page above, the nested <svg> element has `top-margin: 1rem` which is only about 1/8 of the element height, so the click (directed at half of element+margin height) goes well inside the element. So far no idea how to solve the issue properly...
